src/main/java/org/jenkinsci/plugins/mesos/MesosApi.java

```diff
@@ -60,6 +77,10 @@ public MesosApi(String masterUrl, String user, String frameworkName)
    * @return A future reference to the {@link MesosSlave}.
    */
   public CompletableFuture<MesosSlave> startAgent() {
+    PodSpec spec = buildMesosAgentTask(0.1, 32);
+    SpecsSnapshot snapshot = new SpecsSnapshot(convertListToSeq(Arrays.asList(spec)), null);
+    adapter.asAkkaQueues(snapshot, OverflowStrategy.backpressure()).t1();
```

Comment:
I think this has to happen in the constructor. And here we should push `SpecUpdated`. @meln1k, maybe we have to pass a `Source[SpecUpdated]` to the adapter instead. I'm not sure how we could use the `Sink[SpecUpdated]` here.